src/events/leave.js

```diff
@@ -13,6 +14,14 @@ module.exports = function(socket) {
         await Redis.getIO(async (redisIO) => {
             let user = await Users.get(redisIO, userID);
             if (user) {
+                if(user.game) {
```

Comment:
Something is not working when a user decides to leaves the game.

After a game with 4 players, if one chooses to leave, the owner of the room can't start a game.

```
The game (game_26242) winner was user_ZXDUpz5d7aufBHlQAAAB
Reply: OK
New round transaction ok
Reply: OK
Set rematch transaction ok
Rematch set for game game_26242
  have_you_heard Found user id user_ZXDUpz5d7aufBHlQAAAB +28s
Reply: OK
replies = ["OK"]
Rematch transaction ok
User user_ZXDUpz5d7aufBHlQAAAB wants a rematch
Waiting for other players to decide
  have_you_heard Found user id user_RmK-S3xyWozXuyn7AAAB +31s
Reply: OK
replies = ["OK"]
Rematch transaction ok
User user_RmK-S3xyWozXuyn7AAAB wants a rematch
Waiting for other players to decide
  have_you_heard Found user id user_A0WEtD7f0sccG3JMAAAD +2s
Reply: OK
replies = ["OK"]
Rematch transaction ok
User user_A0WEtD7f0sccG3JMAAAD wants a rematch
Waiting for other players to decide
-SK5PPvPleyjS1VLAAAB leave room
Reply: OK
Reply: OK
Game remove player transaction ok
Removed player Nexus S from game game_26242
Reply: OK
Reply: OK
Swap transaction [0]: OK
Swap transaction [1]: OK
User user_-SK5PPvPleyjS1VLAAAB left the room room_26242
Error: Game game_26242 already exists
    at op (/home/yuuma/git/have_you_heard_back/src/server/games.service.js:77:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/yuuma/git/have_you_heard_back/src/events/start.js:54:13
    at async Socket.start_handler (/home/yuuma/git/have_you_heard_back/src/events/start.js:12:9)
Error: Game game_26242 already exists
    at op (/home/yuuma/git/have_you_heard_back/src/server/games.service.js:77:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/yuuma/git/have_you_heard_back/src/events/start.js:54:13
    at async Socket.start_handler (/home/yuuma/git/have_you_heard_back/src/events/start.js:12:9)
Error: Game game_26242 already exists
    at op (/home/yuuma/git/have_you_heard_back/src/server/games.service.js:77:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/yuuma/git/have_you_heard_back/src/events/start.js:54:13
    at async Socket.start_handler (/home/yuuma/git/have_you_heard_back/src/events/start.js:12:9)
Error: Game game_26242 already exists
    at op (/home/yuuma/git/have_you_heard_back/src/server/games.service.js:77:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/yuuma/git/have_you_heard_back/src/events/start.js:54:13
    at async Socket.start_handler (/home/yuuma/git/have_you_heard_back/src/events/start.js:12:9)
Error: Game game_26242 already exists
    at op (/home/yuuma/git/have_you_heard_back/src/server/games.service.js:77:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/yuuma/git/have_you_heard_back/src/events/start.js:54:13
    at async Socket.start_handler (/home/yuuma/git/have_you_heard_back/src/events/start.js:12:9)
Error: Game game_26242 already exists
    at op (/home/yuuma/git/have_you_heard_back/src/server/games.service.js:77:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/yuuma/git/have_you_heard_back/src/events/start.js:54:13
    at async Socket.start_handler (/home/yuuma/git/have_you_heard_back/src/events/start.js:12:9)
Error: Game game_26242 already exists
    at op (/home/yuuma/git/have_you_heard_back/src/server/games.service.js:77:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/yuuma/git/have_you_heard_back/src/events/start.js:54:13
    at async Socket.start_handler (/home/yuuma/git/have_you_heard_back/src/events/start.js:12:9)
Error: Game game_26242 already exists
    at op (/home/yuuma/git/have_you_heard_back/src/server/games.service.js:77:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/yuuma/git/have_you_heard_back/src/events/start.js:54:13
    at async Socket.start_handler (/home/yuuma/git/have_you_heard_back/src/events/start.js:12:9)
Error: Game game_26242 already exists
    at op (/home/yuuma/git/have_you_heard_back/src/server/games.service.js:77:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/yuuma/git/have_you_heard_back/src/events/start.js:54:13
    at async Socket.start_handler (/home/yuuma/git/have_you_heard_back/src/events/start.js:12:9)
Error: Game game_26242 already exists
    at op (/home/yuuma/git/have_you_heard_back/src/server/games.service.js:77:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/yuuma/git/have_you_heard_back/src/events/start.js:54:13
    at async Socket.start_handler (/home/yuuma/git/have_you_heard_back/src/events/start.js:12:9)
User user_ZXDUpz5d7aufBHlQAAAB failed to start the game on room room_26242: Error: Game game_26242 already exists
```

Reply:
I think it is because in removePlayer the user doesn't have a game anymore:
https://github.com/toshisasaki/have_you_heard_back/pull/19/files#diff-f4b65fc34f1f54d3fd10cb18db2ecc984002ea90730021b04a6a17b2df2d1ce8R168